bundle/config/mutator/resolve_variable_references.go

```diff
@@ -215,6 +215,10 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 				// Perform resolution only if the path starts with one of the specified prefixes.
 				for _, prefix := range prefixes {
+					// If the path starts with "artifacts", we need to add a metric to track if this reference is used.
+					if path.HasPrefix(dyn.MustPathFromString("artifacts")) {
```

Comment:
It seems this could be done outside the loop?

Reply:
Good catch! Fixed here https://github.com/databricks/cli/pull/3368/commits/488245e42baee1ab5bcfc28f3327bc424e5467d5